crates/bevy_ecs/src/schedule/system_set.rs

```diff
@@ -145,4 +165,36 @@ impl SystemSet {
         }
         (run_criteria, systems)
     }
+
+    fn sequentialize(systems: &mut Vec<SystemDescriptor>) {
+        let start = NEXT_SEQUENCE_ID.fetch_add(systems.len() as u32, Ordering::Relaxed);
```

Comment:
Why do you use an `AtomicU32` instead of just something local to this system set? How would I add to sequential system sets in parallel?

Reply:
I'm generally against globals (if they can be avoided) so if there is a way to localize this, I would prefer that. 
However, this is a small scope where if there wasn't a way to localize it, that would be ok.

Reply:
The main reason is to generate a compact but effectively globally unique value.

I considered using UUIDs or just random u64s, but there is always a 1/2^n chance of a collision, and while the chance that a collision happens is astronomically small in both cases, a random collision may result seemingly random performance degradation.

This AtomicU32 is both smaller and guarantees that, unless a user adds 2^32 systems to a stage, there is zero chance of a collision. Most consumer machines, and most hosted VMs will likely OOM before that happens. Could try AtomicU16, but I can think of a few use cases to have tens of thousands of systems.

If we localize the u32 to a given instance, we need some point of reference such that other SystemSets are guaranteed not to have an overlap. I currently cannot think of an alternative that affords the same guarantees.

Reply:
Actually, couldn't you just declare a `SequenceId` type local to the method?

Reply:
As discussed on Discord, even if we did that, `SequenceId` would have the same type on every function call.

Reply:
Yea, I'm perfectly ok with the `AtomicU32` solution for now. 
We can't seem to find a better alternative so this should suffice imo.